pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go

```diff
@@ -143,6 +143,11 @@ func (c *dnsRecord) deploy(ctx context.Context, operation string) (extensionsv1a
 		return nil, err
 	}
 
+	// reconcile dnsrecords when the spec changes
+	if !c.values.AnnotateOperation {
+		c.values.AnnotateOperation = c.valuesDontMatchDNSRecord()
```

Comment:
At this point in time, `c.dnsRecord` is not filled with the current spec yet.

You were right, it's probably better to simply combine this with line 152:
```go
		if c.values.AnnotateOperation || c.valuesDontMatchDNSRecord() {
			metav1.SetMetaDataAnnotation(&c.dnsRecord.ObjectMeta, v1beta1constants.GardenerOperation, operation)
		}```

Reply:
Okay!